### PR TITLE
refactor(number): remove value conversion indirection

### PR DESCRIPTION
**Pull request by sno2**
This refactors the old code that used to convert to a value before matching on itself to use the enum variants.

### CODE REVIEW

**Review by aapoalas — APPROVED** [review]
LGTM! Nice cleanup <3

File: nova_vm/src/ecmascript/types/language/number.rs [hunk 1]
```diff
@@ -227,80 +200,68 @@ impl Number {
...
+        match self {
+            Number::Number(n) => {
                 let n = *agent.heap.get(n);
                 n % 1.0 == 0.0 && n % 2.0 == 0.0
             }
-            Value::Integer(n) => Into::<i64>::into(n) % 2 == 0,
-            Value::Float(n) => n % 1.0 == 0.0 && n % 2.0 == 0.0,
-            _ => unreachable!(),
+            Number::Integer(n) => Into::<i64>::into(n) % 2 == 0,
+            Number::Float(n) => n % 1.0 == 0.0 && n % 2.0 == 0.0,
```
[the remark this answers is not in the document]
Comment (sno2) [reply nova_vm/src/ecmascript/types/language/number.rs:225]: True, would you like to PR it?

[the remark this answers is not in the document]
Comment (aapoalas) [reply nova_vm/src/ecmascript/types/language/number.rs:225]: Actually, isn't the test the wrong way around as well? `3 % 2 == 1` right?

[the remark this answers is not in the document]
Comment (sno2) [reply nova_vm/src/ecmascript/types/language/number.rs:225]: huh, yeah. not sure what I was thinking when I wrote that...

**Pull request merged**